server/src/main/java/com/cloud/api/ApiServlet.java

```diff
@@ -170,12 +170,6 @@ void processRequestInContext(final HttpServletRequest req, final HttpServletResp
 
         try {
 
-            if (HttpUtils.RESPONSE_TYPE_JSON.equalsIgnoreCase(responseType)) {
```

Comment:
I think we should keep this, otherwise people won't be able to get XML output if a response type was provided.

Reply:
It's some dummy code. The content type is set when writing the String response to the HttpResponse later on. Doing it so early does not have much sense. See `HttpUtils.writeHttpResponse()`